src/SixLabors.Fonts/Font.cs

```diff
@@ -106,38 +114,81 @@ public Font(Font prototype, float size)
         /// </summary>
         internal FontStyle RequestedStyle { get; }
 
+        /// <summary>
+        /// Gets the filesystem path to the font family source.
+        /// </summary>
+        /// <param name="path">
+        /// When this method returns, contains the filesystem path to the font family source,
+        /// if the path exists; otherwise, the default value for the type of the path parameter.
+        /// This parameter is passed uninitialized.
+        /// </param>
+        /// <returns>
+        /// <see langword="true" /> if the <see cref="Font" /> was created via a filesystem path; otherwise, <see langword="false" />.
+        /// </returns>
+        public bool TryGetPath([NotNullWhen(true)] out string? path)
+        {
+            if (this == default)
+            {
+                FontsThrowHelper.ThrowDefaultInstance();
```

Comment:
I've been using a pattern (when using throw helpers) to help inform the compiler that code doesn't continue here is to mark the helper as returning <T> then adding in a return in despite it never really returning... it helps catch unreachable code etc then too..

Reply:
Yeah I spotted than and wondered by the runtime didn't just do the same rather than introduce an attribute?